base/abstractarraymath.jl

```diff
@@ -449,9 +480,10 @@ See also [`eachrow`](@ref), [`eachcol`](@ref), and [`selectdim`](@ref).
      This function requires at least Julia 1.1.
 """
 @inline function eachslice(A::AbstractArray; dims)
```

Comment:
Would it be possible to have an intermediate step to allow overloading of dispatch on `dims`? Motivated by `dims=:μ` type things in NamedDims.
```suggestion
eachslice(A::AbstractArray; dims) = _eachslice(A::AbstractArray, dims)
@inline function _eachslice(A::AbstractArray, dims)
```